src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -142,6 +143,57 @@ package experimental {
       bs.getWidth
       bs
     }
+
+    /**
+      * Construct a [[BitSet]] matching a range of value
+      *
+      * @param start The smallest matching value
+      * @param length The length of the matching range
+      * @param width The width of the constructed [[BitSet]]. If not given, the returned [[BitSet]] have the width of the maximum possible matching value.
+      * @return A [[BitSet]] matching exactly all inputs in range [start, start + length)
+      */
+    def fromRange(
+      start:  BigInt,
+      length: BigInt,
+      width:  Int = -1
+    ): BitSet = {
```

Comment:
Is `BitSet` the type we want to return here? I'm fine with merging this to master as is but before we release `3.6` I want us to think about if we either want to deprecate `def terms` of `BitSet` or if we want to add a new type above `BitSet` that is not required to be represented using `Set[BitPat]`.

Reply:
I filed #2516 to private `terms` API in 3.6, and #2517 to deprecate in 3.5.
As for the BitSet type, I think in the future(maybe 3.7) it should be a trait to represent BitSet or result of BitSet operations.